workflow/controller/controller.go

```diff
@@ -522,6 +516,63 @@ func (wfc *WorkflowController) runPodCleanup(ctx context.Context) {
 	}
 }
 
+func (wfc *WorkflowController) getPodCleanupPatch(pod *apiv1.Pod, labelPodCompleted bool) ([]byte, error) {
+	un := unstructured.Unstructured{}
+	if labelPodCompleted {
+		un.SetLabels(map[string]string{common.LabelKeyCompleted: "true"})
+	}
+
+	finalizerEnabled := os.Getenv(common.EnvVarPodStatusCaptureFinalizer) == "true"
+	if finalizerEnabled {
+		finalizers, found := make([]string, 0), false
+		for _, f := range pod.Finalizers {
+			if f == common.FinalizerPodStatus {
+				found = true
+			} else {
+				finalizers = append(finalizers, f)
+			}
+		}
+		if found {
```

Comment:
this isn't quite what I meant [before](https://github.com/argoproj/argo-workflows/pull/12862#discussion_r1562025431) -- this is the same as two loops, and two loops are cleaner as you had before.

you can skip the second conditional if you actually use the `index` in `for index, f := range`

```suggestion
		finalizers := slice.RemoveString(pod.Finalizers, common.FinalizerPodStatus)
		if finalizers != pod.Finalizers {
			un.SetFinalizers(finalizers)
```

Reply:
oh, I understand, but it seems we can not compare two slices directlly, so I used pointers, as I can see if no removal the return slice is the same as it passed in.

Reply:
It seems that we can not compare two slices directly, I used `len()` function to do that, I searched it is a field of SliceHeader, so it wouldn't iterate the elements.

Reply:
Ah I meant to compare their pointers (sorry I'm still getting used to Go nuances) since `RemoveString` returns the same slice when no changes. Although I see you tried that and it still didn't work? Odd 🤔

But `len()` is simpler anyway, so that works. Thanks for fixing and checking!